core/src/main/scala/com/gu/tableversions/core/TableVersions.scala

```diff
@@ -13,39 +17,93 @@ trait TableVersions[F[_]] {
     * Start tracking version information for given table.
     * This must be called before any other operations can be performed on this table.
     */
-  def init(table: TableName, isSnapshot: Boolean, userId: UserId, message: UpdateMessage, timestamp: Instant): F[Unit]
+  def init(table: TableName, isSnapshot: Boolean, userId: UserId, message: UpdateMessage, timestamp: Instant): F[Unit] =
+    handleInit(table) {
+      val initialUpdate = TableUpdate(userId, message, timestamp, operations = List(InitTable(table, isSnapshot)))
+      TableState(currentVersion = initialUpdate.header.id, updates = List(initialUpdate))
+    }
 
-  /** Get details about partition versions in a table. */
-  def currentVersion(table: TableName): F[TableVersion]
+  /**
+    * Get details about partition versions in a table.
+    */
+  def currentVersion(table: TableName)(implicit F: Sync[F]): F[TableVersion] =
+    // Derive current version of a table by folding over the history of changes
+    // until either the latest or version marked as 'current' is reached.
+    for {
+      ts <- tableState(table)
+      matchingUpdates = ts.updates.span(_.header.id != ts.currentVersion)
+      updatesForCurrentVersion = matchingUpdates._1 ++ matchingUpdates._2.take(1)
+      operations = updatesForCurrentVersion.flatMap(_.operations)
+    } yield
+      if (isSnapshotTable(operations))
+        latestSnapshotTableVersion(operations)
+      else
+        applyPartitionUpdates(PartitionedTableVersion(Map.empty))(operations)
+
+  /** Return the history of table updates, most recent first. */
+  def updates(table: TableName)(implicit F: Sync[F]): F[List[TableUpdateHeader]] =
+    tableState(table).map(_.updates.map(_.header).reverse)
```

Comment:
This is one of the new operations, getting the version history of a table.

Reply:
Just saw that, re. the point about `updates` being an explicitely sorted structure of sort